### ISSUE TITLE
Accordion Block: Mark icon as `aria-hidden`.

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#71781]
### Description

The accordion toggle icon is a readable character, and is not ignored by screen readers.

The container `.accordion-content__toggle-icon` should have the attribute `aria-hidden="true"` to ensure that it is ignored by screen readers.

### Step-by-step reproduction instructions

1. Create an accordion.
2. Inspect the icon and observe that the icon is not hidden.
3. Traverse the heading with a screen reader using arrow keys for line navigation and observe that the toggle is announced including the '+' sign.

### Environment info

- WordPress trunk
- Gutenberg 21.7.0 RC1

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure
